text/0000-cargo-libstd-awareness.md

```diff
@@ -0,0 +1,236 @@
+The exact format this takes will be determined when custom registries are implemented.
+Based on that, it may be necessary to transform the rust repo before dumping it there.
+
+The "sysroot binary mock registry" is generated by examining the contents of the sysroot.
+Any binary in there will be added to the mock registry, with a version
+taken either from the `language-version` described above, or the compiler version if that key is not present (as it would be with existing rustc releases).
+
+If the compiler source registry exists, that is used, and the sysroot binary mock registry need not even be built.
+If the source registry is absent, then the binary registry is used.
+
+When the build plan just involves the compiler source registry and/or existing types of sources, it can be executed just like today.
+The awkward scenario is when packages from the sysroot binary mock registry need to be used in the build plan.
+Because Cargo doesn't know much about the sysroot binaries, it must be very conservative when deciding whether or not they can be used.
+For example, Cargo may assume they are built with only default features enabled but it can't know what those are.
+If features are explicitly requested, or the default features are disabled (by all dependent packages) then the binaries are ineligible for the build plan being resolved.
+Cargo likewise will have to be conservative with other metadata.
+Packages in the mock binaries registry are not built by Cargo, since they are prebuilt, and when they serve as (transitive) dependencies, Cargo passes them in with `--extern` and their sysroot location.
+This is different from other deps, whose binaries are placed in Cargo's output directory, and sysroot deps today, where `--extern` isn't used as all.
+Since `--extern` is used even in the sysroot binary case, cargo can and will pass `--no-resolve-sysroot` to rustc in all cases.
+
+## Rustbuild improvements
+
+As advertised in the motivation section, with this RFC, rustbuild can use a single workspace to build the standard library and all executables.
+
+One complication with the RFC is that that no sysroot binaries or source associated with the bootstrap compiler (or previous stage) are ever used; one needs to bypass the compiler source registry and sysroot binary mock registry.
+To accomplish this, rustbuild's workspace will need to use `[replace]` to redirect all stdlib deps to use the workspace's packages.
+
+All binaries for a specific phase can be built with a single `cargo build` (barring special requirements for individual libraries).
+Rather than have a multitude of build artifact directories per stage, only one is needed.
+After the last compiler is build, an additional mini-stage of building just the standard library could be performed, but distributions wishing to build all deps from source in a standardized fashion (e.g. probably NixOS) would forgo this.
```

Comment:
This paragraph seems pretty incidental to the RFC and I don't follow it well. Can it be removed?